promote-release-to-box.py

```diff
@@ -0,0 +1,290 @@
+import os
+import re
+import sys
+import json
+import yaml
+import argparse
+import requests
+import logging
+import subprocess
+from tqdm import tqdm
+from boxsdk import OAuth2,Client
+from artifactory import ArtifactoryPath
+from requests.auth import HTTPBasicAuth
+from boxsdk.exception import BoxAPIException
+
+
+logging.basicConfig(
+     level=logging.WARNING, 
+     format= '[%(asctime)s] {%(pathname)s:%(lineno)d} %(levelname)s - %(message)s',
+     datefmt='%H:%M:%S'
+ )
+
+def get_artifact_info_json(build_name, build_number, rt_auth = (None, None), rt_base_url = None):
+    ''' 
+    Expected Jfrog CLI is availble in the system.
+    Executes these:
+    1. jf config add orl-artifactory --interactive=false \
+        --enc-password=false --basic-auth-only \
+        --artifactory-url https://repo.orl.eng.hitachivantara.com/artifactory \
+        --password --user buildguy
+
+    2. jf rt search --server-id orl-artifactory \
+        --props "build.name=pdi-xxx-9.5.1.0;build.number=86" "*-9.5.1.0-86.zip" \
+        > builds.json
+
+    Alternative way: Use ArtifactoryBuildManager https://github.com/devopshq/artifactory#builds
+    
+    ^^^^^^^^^^^^^
+    from artifactory import ArtifactoryBuildManager
+
+    arti_build = ArtifactoryBuildManager(
+        orl_url, auth=auth_orl)
+
+    # Get all builds,space turns into %
+    all_builds = arti_build.builds
+    print(all_builds)
+    ^^^^^^^^^^^^^
+
+    '''
+    
+
+    # Define the command and arguments
+    command = [
+        'jf', 'config', 'add', 'orl-artifactory',
+        '--interactive=false', '--enc-password=false', '--basic-auth-only',
+        '--artifactory-url', f'{rt_base_url}',
+        '--password', f'{rt_auth[0]}',
+        '--user', f'{rt_auth[1]}'
+    ]
```

Comment:
Maybe the artifactory config can be some argument that is passed in since [in the workflow](https://github.com/pentaho/actions-common/pull/1/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R70-R74) an artifactory is already being configured. If not passed in, then configure it. That input arg can then be used in the `jf rt search...`.

Reply:
Sure, I could just remove the configuration of artifactory part and let workflow file handle it.

Reply:
I wouldn't remove it entirely. This may be reused somewhere else where there isn't a previous artifactory config, therefore, there might be the need to config it in the script.